library/x509_crt.c

```diff
@@ -3012,23 +3144,51 @@ static int x509_crt_check_cn( const mbedtls_x509_buf *name,
     return( -1 );
 }
 
+static int x509_crt_check_san_ip( const mbedtls_x509_sequence *san,
+                                  const char *cn, size_t cn_len )
+{
+    uint32_t ip[4];
+    cn_len = mbedtls_x509_crt_parse_cn_inet_pton( cn, ip );
+    if( cn_len == 0 )
+        return( -1 );
+
+    for( const mbedtls_x509_sequence *cur = san; cur != NULL; cur = cur->next )
+    {
+        const unsigned char san_type = (unsigned char) cur->buf.tag &
+                                       MBEDTLS_ASN1_TAG_VALUE_MASK;
+        if( san_type == MBEDTLS_X509_SAN_IP_ADDRESS &&
+            cur->buf.len == cn_len && memcmp( cur->buf.p, ip, cn_len ) == 0 )
+            return( 0 );
+    }
+
+    return( -1 );
+}
+
 /*
  * Check for SAN match, see RFC 5280 Section 4.2.1.6
  */
-static int x509_crt_check_san( const mbedtls_x509_buf *name,
+static int x509_crt_check_san( const mbedtls_x509_sequence *san,
                                const char *cn, size_t cn_len )
 {
-    const unsigned char san_type = (unsigned char) name->tag &
-                                   MBEDTLS_ASN1_TAG_VALUE_MASK;
-
-    /* dNSName */
-    if( san_type == MBEDTLS_X509_SAN_DNS_NAME )
-        return( x509_crt_check_cn( name, cn, cn_len ) );
-
-    /* (We may handle other types here later.) */
+    int san_ip = 0;
+    for( const mbedtls_x509_sequence *cur = san; cur != NULL; cur = cur->next )
+    {
+        switch( (unsigned char) cur->buf.tag & MBEDTLS_ASN1_TAG_VALUE_MASK )
+        {
+        case MBEDTLS_X509_SAN_DNS_NAME:                    /* dNSName */
+            if( x509_crt_check_cn( &cur->buf, cn, cn_len ) == 0 )
+                return( 0 );
+            break;
+        case MBEDTLS_X509_SAN_IP_ADDRESS:                  /* iPAddress */
+            san_ip = 1;
```

Comment:
```suggestion
                san_ip = 1;
```

Reply:
not done.  doing so would violate the coding guidelines.  please clarify your intent.